src/hooks/use-lesson.ts

```diff
@@ -98,10 +95,8 @@ export const useLesson = (): UseLesson => {
   }, [lessonId]);
 
   useEffect(() => {
-    if (courseId && course.id !== Number(courseId)) {
-      void dispatch(fetchCourseById(courseId));
-    }
-  }, [course, courseId]);
+    void dispatch(fetchCourseById(courseId));
+  }, [lessonProcess]);
```

Comment:
Статусов процесса загрузки 4 штуки (ProcessEnum). Запрос за курсом будет вызываться как минимум 3 раза при первом открытии страницы урока и по 2 раза при переходах на другие уроки.

Нужно запрашивать актуальное содержание только после того, как загрузится урок:

```suggestion
    if (courseId && lessonProcess === ProcessEnum.Succeeded) {
      void dispatch(fetchCourseById(courseId));
    }
  }, [lessonProcess, courseId]);
```

Reply:
Добавил условие для вызова запроса курса